internal/pkg/artifact/download/http/downloader.go

```diff
@@ -12,18 +12,32 @@ import (
 	"net/url"
 	"os"
 	"path"
+	"strconv"
 	"strings"
+	"time"
+
+	"github.com/docker/go-units"
 
 	"github.com/elastic/beats/v7/libbeat/common/transport/httpcommon"
-	"github.com/elastic/elastic-agent/internal/pkg/agent/program"
+	"github.com/elastic/elastic-agent-libs/atomic"
 
 	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/program"
 	"github.com/elastic/elastic-agent/internal/pkg/artifact"
 	"github.com/elastic/elastic-agent/internal/pkg/release"
 )
 
 const (
 	packagePermissions = 0660
+
+	// Reports the current download progress when percentage of time has passed in the overall interval for the
+	// complete download to complete. 5% is a good default, as the default timeout is 10 minutes and this will have it
+	// log every 30 seconds.
+	downloadProgressIntervalPercentage = 0.05
```

Comment:
[Go convention]

```suggestion
	// downloadProgressIntervalPercentage defines how often to report the current download progress when percentage of time has passed in the overall interval for the
	// complete download to complete. 5% is a good default, as the default timeout is 10 minutes and this will have it
	// log every 30 seconds.
	downloadProgressIntervalPercentage = 0.05
```